Assets/MixedRealityToolkit-Preview/LightEstimation/Scripts/LightCapture.cs

```diff
@@ -0,0 +1,256 @@
+	[SerializeField]              float stampExpireDistance = 0;
+
+	[Header("Directional Lighting")]
+	[Tooltip("Should the system calculate information for a directional light? This will scrape the lower mips of the Cubemap to find the direction and color of the brightest values, and apply it to the scene's light.")]
+	[SerializeField]             bool  useDirectionalLight       = true;
+	[Tooltip("When finding the primary light color, it will average the brightest 20% of the pixels, and use that color for the light. This sets the cap for the saturation of that color.")]
+	[SerializeField, Range(0,1)] float maxLightColorSaturation   = 0.3f;
+	[Tooltip("The light eases into its new location when the information is updated. This is the speed at which it eases to its new destination, measured in degrees per second.")]
+	[SerializeField]             float lightAngleAdjustPerSecond = 45f;
+
+	[Header("Optional Overrides")]
+	[Tooltip("Defaults to Camera.main. Which object should we be looking at for our orientation and position?")]
+	[SerializeField] Transform       cameraOrientation;
+	[Tooltip("Default will pick from the scene, or create one automatically. If you have settings you'd like to configure on your probe, hook it in here.")]
+	[SerializeField] ReflectionProbe probe;
+	[Tooltip("Default will pick the first directional light in the scene. If no directional light is found, one will be created!")]
+	[SerializeField] Light           directionalLight;
+	
+	ICameraCapture captureCamera;
+	CubeMapper     map = null;
+	Histogram      histogram = new Histogram();
+	Material       startSky;
+	int            stampCount;
+	Texture2D      tex;
+
+	// For easing the light directions
+	Quaternion lightTargetDir;
+	Quaternion lightStartDir;
+	float      lightStartTime = -1;
+	float      lightTargetDuration;
+
+	public CubeMapper CubeMapper { get { return map; } }
+	#endregion
+
+	#region Unity Events
+	private void Awake ()
+	{
+		// Pick camera based on platform, PhotoCaptureCamera has camera controls,
+		// but doesn't use front camera on PC like WebcamCamera does.
+		
+		#if WINDOWS_UWP && !UNITY_EDITOR
+		captureCamera = new CameraCaptureUWP();
+		#elif (UNITY_ANDROID || UNITY_IOS) && !UNITY_EDITOR
+		_camera = new CameraCaptureARFoundation();
+		#else
+		captureCamera = new CameraCaptureWebcam(Camera.main.transform, Camera.main.fieldOfView);
+		#endif
+
+		// Make sure we have access to a probe in the scene
+		if (probe == null) 
+			probe = FindObjectOfType<ReflectionProbe>();
+		if (probe == null)
+		{
+			GameObject probeObj = new GameObject("_LightCaptureProbe", typeof(ReflectionProbe));
+			probeObj.transform.SetParent(transform, false);
+			probe = probeObj.GetComponent<ReflectionProbe>();
+			
+			probe.size          = Vector3.one * 10000;
+			probe.boxProjection = false;
+		}
+
+		// Same with a camera object
+		if (cameraOrientation == null)
+			cameraOrientation = Camera.main.transform;
```

Comment:
Nit: It looks like you are trying to minimize dependencies on the MRTK, but it might be nice to use the CameraCache for consistency.